File: src/main/java/com/summer/lijiahao/base/ConfigureFileUtil.java
package com.summer.lijiahao.base;

import java.io.*;
import java.net.URL;
import java.nio.charset.StandardCharsets;

public class ConfigureFileUtil {

    /**
     * 读取数据库模版文件
     *
     * @param fileName
     * @return
     */
    public String getVMFile(String fileName) {
        return this.getClass().getResource("../../.." + fileName).getFile();
    }

    /**
     * 读取输入模版
     *
     * @param fileName
     * @return
     */
    public String readTemplate(String fileName) throws BusinessException {
        URL test = this.getClass().getResource("/template/" + fileName);
        InputStream in = this.getClass().getResourceAsStream("/template/" + fileName);
        return readTemplate(in);
    }

    /**
     * 读取输入模板
     *
     * @param file
     * @return
     * @throws BusinessException
     */
    public String readTemplate(File file) throws BusinessException {
        try {
            FileInputStream fileInputStream = new FileInputStream(file);
            return readTemplate(fileInputStream);
        } catch (FileNotFoundException e) {
//            e.printStackTrace();
            throw new BusinessException(e.getMessage());
        }
    }

    private String readTemplate(InputStream in) throws BusinessException {
        StringBuilder tempBuilder = new StringBuilder();
        try {
            InputStreamReader isr = new InputStreamReader(in, StandardCharsets.UTF_8);
            BufferedReader br = new BufferedReader(isr);
            String lineTxt;
            while (true) {
                if ((lineTxt = br.readLine()) == null) break;
                tempBuilder.append(lineTxt);
                tempBuilder.append("\r\n");
            }
            br.close();
        } catch (IOException e) {
//            e.printStackTrace();
            throw new BusinessException(e.getMessage());
        }
        return tempBuilder.toString();
    }

    /**
     * 输出文件
     *
     * @param file
     * @param content
     */
    public void outFile(File file, String content, String charset, boolean bomFlag) {
        try {
            FileOutputStream fos = new FileOutputStream(file);
            OutputStreamWriter dos = null;
            if (bomFlag) {
                fos.write(new byte[]{(byte) 0xEF, (byte) 0xBB, (byte) 0xBF});
            }
            dos = new OutputStreamWriter(fos, charset);
            dos.write(content);
            dos.close();
        } catch (FileNotFoundException e) {
//            e.printStackTrace();
        } catch (IOException e) {
//            e.printStackTrace();
        }
    }
}

File: src/main/kotlin/com/summer/lijiahao/nccdevtool/base/util/ConfigureFileUtil.kt
package com.summer.lijiahao.nccdevtool.base.util

import java.io.*
import java.nio.charset.StandardCharsets

class ConfigureFileUtil {

    fun readTemplate(fileName: String): String {
        val inputStream = this.javaClass.getResourceAsStream("/template/$fileName")
        return readTemplate(inputStream)
    }

    fun readTemplate(file: File?): String {
        return try {
            val fileInputStream = FileInputStream(file)
            readTemplate(fileInputStream)
        } catch (e: FileNotFoundException) {
            throw FileNotFoundException(e.message)
        }
    }

    private fun readTemplate(inputStream: InputStream): String {
        val tempBuilder = StringBuilder()
        try {
            val isr = InputStreamReader(inputStream, StandardCharsets.UTF_8)
            val br = BufferedReader(isr)
            var lineTxt: String?
            while (true) {
                if (br.readLine().also { lineTxt = it } == null) break
                tempBuilder.append(lineTxt)
                tempBuilder.append("\r\n")
            }
            br.close()
        } catch (e: IOException) {
            throw IOException(e.message)
        }
        return tempBuilder.toString()
    }

    fun outFile(file: File?, content: String?, charset: String?, bomFlag: Boolean) {
        try {
            val fos = FileOutputStream(file)
            var dos: OutputStreamWriter? = null
            if (bomFlag) {
                fos.write(byteArrayOf(0xEF.toByte(), 0xBB.toByte(), 0xBF.toByte()))
            }
            dos = OutputStreamWriter(fos, charset)
            dos.write(content)
            dos.close()
        } catch (_: FileNotFoundException) {
        } catch (_: IOException) {
        }
    }
}